### PR TITLE
Enabled and fixed a number of lints

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -11,42 +11,62 @@ analyzer:
     dead_code: error
 linter:
   rules:
-    # Errors
-    - avoid_empty_else
+    - avoid_catching_errors
```
Comment (kevmoo) [inline analysis_options.yaml:14]: FYI: the deleted lints are already covered by pkg:pedantic 1.9

**Comment by kevmoo** [comment]
CC @devoncarew as FYI

File: lib/src/common.dart [hunk 2]
```diff
@@ -70,14 +70,14 @@ String normalizeProjectName(String name) {
 /// A key value can only be an ASCII string made up of letters: A-Z, a-z.
 /// No whitespace, numbers, or other characters are allowed.
 String substituteVars(String str, Map<String, String> vars) {
-  var nonValidKeys =
+  final nonValidKeys =
       vars.keys.where((k) => k.contains(_nonValidSubstituteRegExp)).toList();
   if (nonValidKeys.isNotEmpty) {
     throw ArgumentError('vars.keys can only contain letters.');
   }
```
Comment (parlough) [inline lib/src/common.dart:77]: I know you didn't write this, but seems appropriate to clean it up here. If not, that's fine as well.

Seems `nonValidKeys` is only used for this check, so could change this whole chunk to simply:

```dart
if (vars.keys.any((k) => k.contains(_nonValidSubstituteRegExp))) {
  throw ArgumentError('vars.keys can only contain letters');
}
```

**Review by parlough — APPROVED** [review]
Lots of good cleanup here 😁

File: lib/src/common.dart (hunk 2 above)
Comment (kevmoo) [reply lib/src/common.dart:77]: solid idea!

Comment (kevmoo) [reply lib/src/common.dart]: done!

**Review by kwalrath — APPROVED** [review]
LGTM, although I'd feel better if @domesticmouse took a look too.

**Review by domesticmouse — APPROVED** [review]
LGTM

**Pull request merged**